pkg/blobinfocache/internal/prioritize/prioritize.go

```diff
@@ -25,16 +25,26 @@ const replacementAttempts = 5
 // This is a heuristic/guess, and could well use a different value.
 const replacementUnknownLocationAttempts = 2
 
-// CandidateCompression returns (true, compressionOp, compressionAlgo) if a blob
-// with compressionName (which can be Uncompressed or UnknownCompression) is acceptable for a CandidateLocations* call with v2Options.
+// CandidateTemplate is a subset of BICReplacementCandidate2 with data related to a specific digest,
+// which can be later combined with information about a location.
+type CandidateTemplate struct {
+	digest               digest.Digest
+	compressionOperation types.LayerCompression // Either types.Decompress for uncompressed, or types.Compress for compressed
+	compressionAlgorithm *compression.Algorithm // An algorithm when the candidate is compressed, or nil when it is uncompressed
+}
+
+// CandidateTemplateWithCompression returns a CandidateTemplate if a blob with compressionName (which can be Uncompressed
+// or UnknownCompression) is acceptable for a CandidateLocations* call with v2Options.
 //
 // v2Options can be set to nil if the call is CandidateLocations (i.e. compression is not required to be known);
 // if not nil, the call is assumed to be CandidateLocations2.
-//
-// The (compressionOp, compressionAlgo) values are suitable for BICReplacementCandidate2
-func CandidateCompression(v2Options *blobinfocache.CandidateLocations2Options, digest digest.Digest, compressorName string) (bool, types.LayerCompression, *compression.Algorithm) {
+func CandidateTemplateWithCompression(v2Options *blobinfocache.CandidateLocations2Options, digest digest.Digest, compressorName string) *CandidateTemplate {
 	if v2Options == nil {
-		return true, types.PreserveOriginal, nil // Anything goes. The (compressionOp, compressionAlgo) values are not used.
+		return &CandidateTemplate{ // Anything goes. The CompressionOperation, CompressionAlgorithm values are not used.
```

Comment:
1. If `compression{Type,Algorithm}` fields are not used, it does not make sense to initialize them explicitly -- we can just omit those and default values will be assigned by Go (those seems to be `nil` for algo and `types.PreserveOriginal` for op).

2. nit: `s/Compression/compression/`.

Reply:
Fixed. Thanks!